File: client/src/app/_dtos/admin/orderItemsAndAgentsToFwdDto.ts

import { ICustomerOfficialDto } from "src/app/_models/admin/customerOfficialDto";
import { IOrderItemToFwdDto } from "./orderItemToFwdDto";

export interface IOrderItemsAndAgentsToFwdDto
{
     orderId: number;
     orderNo: number;
     orderDate: Date;
     customerId: number;
     customerName: string;
     projectManagerId: number;
     items: IOrderItemToFwdDto[];
     agents: ICustomerOfficialDto[];
     dateForwarded: Date;
}

export class OrderItemsAndAgentsToFwdDto implements IOrderItemsAndAgentsToFwdDto
{
     orderId= 0;
     orderNo= 0;
     orderDate = new Date('1900-01-01');
     customerId= 0;
     customerName= '';
     projectManagerId= 0;
     items: IOrderItemToFwdDto[]=[];
     agents: ICustomerOfficialDto[]=[];
     dateForwarded = new Date('1900-01-01');
}

File: api/DTOs/Admin/Orders/OrderItemsAndAgentsToFwdDto.cs
using api.Entities.Admin;

namespace api.DTOs.Admin.Orders
{
    public class OrderItemsAndAgentsToFwdDto
    {
        public int OrderId { get; set; }
        public int OrderNo { get; set; }
        public DateTime OrderDate { get; set; }
        public int CustomerId { get; set; }
        public string CustomerName { get; set; }
        public string AboutEmployer { get; set; }
        public int ProjectManagerId {get; set;}
        public ICollection<OrderItemToFwdDto> Items {get; set;}
        public ICollection<AssociatesToFwdDto> Agents {get; set;}
        public DateTime DateForwarded {get; set;}
    }
}
